execution_workflows/APAtrap/subworkflows/run_differential.nf

```diff
@@ -0,0 +1,60 @@
+/*
+    Run APAtrap to obtain differential challenge output
+*/
+
+include { IDENTIFY_DISTAL_3UTR  } from '../modules/identify_distal_3utr' addParams( options: [:] )
+include { PREDICT_APA           } from '../modules/predict_apa' addParams( options: [:] )
+include { CONVERT_TO_BED        } from '../modules/convert_to_bed' addParams( options: [:] )
+include { DE_APA                } from '../modules/de_apa' addParams( options: [:] )
+include { CONVERT_TO_TSV        } from '../modules/convert_to_tsv' addParams( options: [:] )
+
+workflow RUN_DIFFERENTIAL {
+    take:
+    ch_preprocessing_input
+    ch_sample_bedgraph_files_dir
+    ch_3utr_input
+
+    main:
+
+    ch_sample_bedgraph_files_dir
+        .first()
+        .set{ ch_sample_bedgraph_files_dir }
+
+    /*
+        Perform first step of APAtrap: identify distal 3'utr
+    */
+    IDENTIFY_DISTAL_3UTR( ch_sample_bedgraph_files_dir,
+                          ch_3utr_input )
+    // combine the bedgraph read with the inputfile for predictAPA
+    ch_3utr_input
+        .map{ it -> [ it[1], it[2] ] }
+        .combine(IDENTIFY_DISTAL_3UTR.out.ch_predictapa_input, by:0)
+        .unique()
+        .set { ch_predict_apa_input }
+
+    ch_predict_apa_input
+        .first()
+        .set{ ch_predict_apa_input }
+
+    /*
+        Perform second step of APAtrap: infer all potential APA sites
+    */
+    PREDICT_APA( ch_sample_bedgraph_files_dir,
+                ch_predict_apa_input )
```

Comment:
This seems to be the main part of identification and quantification. 
Does this mean that when running only differential it would run ident. and quant.? 
And would I be able to run differential from already present results? Based on the test data and setting all 3 to `true`, identification & quantification and differential run separately.

I guess we would want to run benchmarks from the same input files anyway, correct?

Reply:
Good point! So running differential would require identification and quantification to run too but in a different command compared to when we're running identification and/or quantificiation only. This is because when running differential, we want samples of the same condition to be grouped together for indentification and quantification. On the other hand, without running differential, we want each sample to be ran individually no matter the condition. The difference in the commands are found inside the module (e.g. https://github.com/iRNA-COSI/APAeval/blob/main/execution_workflows/APAtrap/modules/predict_apa.nf). This difference in commands is why when we set all three settings to be true, we have to re-run identification and quantification when running differential instead of runinng differential from the already present results